src/garage/_dtypes.py

```diff
@@ -428,22 +431,29 @@ def next_observations(self):
             ]))
 
 
+class StepType(enum.IntEnum):
```

Comment:
when you introduce this, you should also modify `TimeStepBatch` and `TrajectoryBatch` to use it.